pkg/workflow/step/service_step.go

```diff
@@ -31,69 +72,53 @@ commonPath="$serverURL/v1/projects/$projectID/environments/$environmentID"
-	name=$(echo $response | jq -r '.name')
+    response=$(curl -sS "$commonPath/resources" -X "POST" -H "content-type: application/json" -H "Authorization: Bearer $token" -d '{{inputs.parameters.attributes}}' $tlsVerify)
+    name=$(echo $response | jq -r '.name')
 
-	if [ "$name" == "null" ]; then
-		echo "failed create resource, response: $response"
-		exit 1
-	fi
+    if [ "$name" == "null" ]; then
+        echo "failed create resource, response: $response"
+        exit 1
+    fi
 fi
 
 # If jobType upgrade resource.
 if [ "$jobType" == "upgrade" ]; then
-	status=$(curl -sS -w "%{http_code}" -o /dev/null "$commonPath/resources/$resourceName/upgrade" -X "PUT" -H "content-type: application/json" -H "Authorization: Bearer $token" -d '{{inputs.parameters.attributes}}' $tlsVerify)
+    status=$(curl -sS -w "%{http_code}" -o /dev/null "$commonPath/resources/$resourceName/upgrade" -X "PUT" -H "content-type: application/json" -H "Authorization: Bearer $token" -d '{{inputs.parameters.attributes}}' $tlsVerify)
 
-	# if not status >= 200 and status < 300
-	if [ "$status" -lt 200 ] || [ "$status" -ge 300 ]; then
-		echo "failed upgrade resource, response status: $status"
-		exit 1
-	fi
+    # if not status >= 200 and status < 300
+    if [ "$status" -lt 200 ] || [ "$status" -ge 300 ]; then
+        echo "failed upgrade resource, response status: $status"
+        exit 1
+    fi
 fi
 
 # Get latest run id
-runResponse=$(curl -sS "$commonPath/resources/$resourceName/runs?page=1&perPage=1&sort=-createTime" -X GET -H "Authorization: Bearer $token" $tlsVerify)
+runResponse=$(common_get "$commonPath/resources/$resourceName/runs?page=1&perPage=1&sort=-createTime")
 runID=$(echo $runResponse | jq -r '.items[0].id')
 
-# Watch service logs until the service finished.
-curl -o - -sS "$commonPath/resources/$resourceName/runs/$runID/log?jobType=apply&watch=true" -X GET -H "Authorization: Bearer $token" $tlsVerify --compressed
-
-# Check run status. wait until run status is ready.
-timeout=30
-factor=1
-while true; do
-	statusResponse=$(curl -sS "$commonPath/resources/$resourceName/runs/$runID" -X GET -H "Authorization: Bearer $token" $tlsVerify)
-	statusSummary=$(echo $statusResponse | jq -r '.status.summaryStatus')
-
-	if [ "$statusSummary" == "Succeeded" ]; then
-		break
-	fi
-
-	# If status is failed, exit.
-	if [ "$statusSummary" == "Failed" ]; then
-		exit 1
-	fi
-
-	if [ "$timeout" -le 0 ]; then
-		exit 1
-	fi
-
-	sleep $((factor * 2))
-	factor=$((factor * 2))
-	timeout=$((timeout - factor))
-done
+# Watch run plan logs until the plan finished.
+common_get "$commonPath/resources/$resourceName/runs/$runID/log?jobType=plan&watch=true" --compressed
+
+# Check run status, wait until run status is running.
+wait_run_status "$resourceName" "$runID" "Running"
+
+# Watch run logs until the run finished.
+common_get "$commonPath/resources/$resourceName/runs/$runID/log?jobType=apply&watch=true" --compressed
```

Comment:
Is the compressed arg valid? Looks like common_get is not handling it.

Reply:
updated.